azure/services/subnets/subnets.go

```diff
@@ -72,10 +72,10 @@ func (s *Service) Reconcile(ctx context.Context) error {
 				continue
 			}
 
-			subnet.Role = subnetSpec.Role
-			subnet.Name = existingSubnet.Name
-			subnet.CIDRBlocks = existingSubnet.CIDRBlocks
 			subnet.ID = existingSubnet.ID
+			subnet.CIDRBlocks = existingSubnet.CIDRBlocks
+
+			s.Scope.SetSubnet(subnet)
```

Comment:
This is being called from inside a loop going through the subnets, then it does a loop again, might be better to pass in the name or something like that and just updating that subnet, what do you think?

Reply:
the issue with this is that we don't have an efficient way to Get() a particular subnet in O(1)... that's what I attempted to fix in #1280 but it's not best practices for CRDs to use maps so we decided to not move forward with it. Even if we pass in a name we still have to loop through the subnets to find it. And if we were to collect a list of subnets to update and then make one call to SetSubnets() we'd still have a nested for loop to loop through the current subnets and the desired updates and compare names. We _could_ optionally store this "desired updates" list as a map and pass that into SetSubnets() but that's not going to bring that much optimization given that the number of subnets is a constant 2.

Reply:
Taking into account that the number of subnets won't be a big number, I'd say it's fine. Going for a more complicated solution could harm the readability with not real gains in performance (again, due to the fact that usually we expect a low number of subnets).